<?xml version="1.0"?>
<documentation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="https://phpcsstandards.github.io/PHPCSDevTools/phpcsdocs.xsd"
    title="Test Doubles"
    >
    <standard>
    <![CDATA[
    Double/Mock test helper OO structures (classes/interfaces/traits/enums) should be in their own file and placed in a dedicated test fixtures sub-directory.

    The name of the dedicated sub-directory is configurable.
    ]]>
    </standard>
    <standard>
    <![CDATA[
    Double/Mock test helper classes should have "Double" or "Mock" in their class name.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Test double suffixed with 'Double'.">
        <![CDATA[
class <em>Test_Double</em> extends Original_Class {
    // Code.
}
        ]]>
        </code>
        <code title="Invalid: Test double without 'Double' or 'Mock' in the class name.">
        <![CDATA[
class <em>Foo_Class<em> extends Original_Foo_Class {
    // Code.
}
        ]]>
        </code>
    </code_comparison>
</documentation>
