<?xml version="1.0"?>
<documentation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="https://phpcsstandards.github.io/PHPCSDevTools/phpcsdocs.xsd"
    title="Test Doubles"
    >
    <standard>
    <![CDATA[
    Double/Mock test helper classes should be in their own file and placed in a dedicated test doubles sub-directory.
The name of the dedicated sub-directory is configurable.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Double class in separate file in double sub-directory.">
        <![CDATA[
<!-- Doubles file in double sub-directory -->
<?php
class Test_Double extends Original_Class {
    // Code.
}

<!-- Unit test file -->
<?php
class Test_Original_Class extends TestCase {
    // Test code.
}
        ]]>
        </code>
        <code title="Invalid: Having both the double class as well as the test class in the same file in the test directory.">
        <![CDATA[
<!-- Unit test file -->
<?php
class Test_Double extends Original_Class {
    // Code.
}

class Test_Original_Class extends TestCase {
    // Test code.
}
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
    Double/Mock test helper classes should have "Double" or "Mock" in their class name.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Test double suffixed with 'Double'.">
        <![CDATA[
class <em>Test_Double</em> extends Original_Class {
    // Code.
}
        ]]>
        </code>
        <code title="Invalid: Test double without 'Double' or 'Mock' in the class name.">
        <![CDATA[
class <em>Foo_Class<em> extends Original_Foo_Class {
    // Code.
}
        ]]>
        </code>
    </code_comparison>
</documentation>
